pkg/scheduler/cache/util.go

```diff
@@ -29,11 +32,55 @@ func CreateNodeNameToInfoMap(pods []*v1.Pod, nodes []*v1.Node) map[string]*NodeI
 		}
 		nodeNameToInfo[nodeName].AddPod(pod)
 	}
+	imageStatesMap := createNodeImageStatesMap(nodes)
+
 	for _, node := range nodes {
 		if _, ok := nodeNameToInfo[node.Name]; !ok {
 			nodeNameToInfo[node.Name] = NewNodeInfo()
 		}
-		nodeNameToInfo[node.Name].SetNode(node)
+		nodeInfo := nodeNameToInfo[node.Name]
+		nodeInfo.imageStates = imageStatesMap[node.Name]
+		nodeInfo.SetNode(node)
```

Comment:
Got it, will fix.

Reply:
Just in case I miss important point - I understand the rationale here being "it makes more sense to set node and then add imageStates" but correctness-wise shouldn't they be equivalent?

Reply:
Yes, they are equivalent in the current implementation.